cmd/ulxly/bridge_service/agglayer/service.go

```diff
@@ -0,0 +1,262 @@
+	// 	return nil, 0, bridge_service.ErrNotFound
+	// }
+
+	// bridgesResponses := make([]bridgeResponse, 0, limit)
+	// bridgesResponses = append(bridgesResponses, resp.Bridges[skipItems:pageSize]...)
+
+	// // loads the remaining part of deposits when offset is not exactly the size of a page
+	// // this is needed because the API only supports pagination by page number and page size
+	// // and not by offset and limit
+	// if skipItems > 0 {
+	// 	endpoint := fmt.Sprintf(endpointTemplate, s.BridgeServiceBase.Url(), urlPath, destinationAddress, pageNumber+1, pageSize)
+	// 	resp, respError, statusCode, err = httpjson.HTTPGetWithError[getBridgesResponse, errorResponse](s.httpClient, endpoint)
+	// 	if err != nil {
+	// 		return nil, 0, err
+	// 	}
+
+	// 	if statusCode != http.StatusOK {
+	// 		errMsg := "unable to retrieve bridge deposits"
+	// 		log.Warn().Int("code", statusCode).Str("message", respError.Error).Msgf("%s", errMsg)
+	// 		return nil, 0, bridge_service.ErrNotFound
+	// 	}
+
+	// 	end := skipItems
+	// 	if end > len(resp.Bridges) {
+	// 		end = len(resp.Bridges)
+	// 	}
+
+	// 	bridgesResponses = append(bridgesResponses, resp.Bridges[0:end]...)
+	// }
+
+	// deposits := make([]bridge_service.Deposit, 0, len(bridgesResponses))
+	// for _, bridgeResp := range bridgesResponses {
+	// 	networkID := 0
+	// 	if bridgeResp.DestNet
+
+	// 	deposit, err := s.responseToDeposit(bridgeResp)
+	// 	if err != nil {
+	// 		return nil, 0, err
+	// 	}
+	// 	deposits = append(deposits, *deposit)
+	// }
+
+	// return deposits, resp.Count, nil
+	panic("not implemented yet") // bridges endpoint requires network_id parameter
+}
+
+func (s *BridgeService) GetProof(depositNetwork, depositCount uint32, ger *common.Hash) (*bridge_service.Proof, error) {
+	var l1InfoTreeIndex uint32
+
+	if ger != nil {
+		return nil, errors.New("getting proof by ger is not supported yet by Agglayer bridge service")
+	}
+
+out:
+	for {
+		select {
+		case <-time.After(time.Minute):
+			return nil, fmt.Errorf("timeout waiting for l1 info tree index")
+		default:
+			idx, err := s.getL1InfoTreeIndex(depositNetwork, depositCount)
+			if errors.Is(err, bridge_service.ErrNotFound) {
+				time.Sleep(time.Second)
+				continue
+			}
+			if err != nil {
+				return nil, err
+			}
+
+			l1InfoTreeIndex = *idx
+			break out
+		}
+	}
+
+	endpoint := fmt.Sprintf("%s/%s/claim-proof?network_id=%d&leaf_index=%d&deposit_count=%d", s.BridgeServiceBase.Url(), urlPath, depositNetwork, l1InfoTreeIndex, depositCount)
+	resp, respError, statusCode, err := httpjson.HTTPGetWithError[getClaimProofResponse, errorResponse](s.httpClient, endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if statusCode != http.StatusOK {
+		if statusCode == http.StatusNotFound {
+			return nil, bridge_service.ErrNotFound
+		}
+		errMsg := "unable to retrieve proof"
+		log.Warn().Int("code", statusCode).Str("message", respError.Error).Msgf("%s", errMsg)
+		return nil, fmt.Errorf(respError.Error)
```

Comment:
Normally I would prefer to wrap..
```suggestion
		return nil, fmt.Errorf("unable to get proof: %w", respError.Error)
```

Assuming that `respError.Error` is an `error`

Reply:
- `respError` is a `DTO` to unmarshal error responses.
- `respError.Error` is just a string
- we can achieve wrapping by doing this `return nil, fmt.Errorf("unable to get proof: %w", fmt.Errorf(respError.Error))`
- wrapping it in this case doesn't help too much, because we don't have an error type to check against it in the upper levels.
- IMO the best for now is `return nil, fmt.Errorf("unable to get proof: %s", respError.Error)`